File: android/src/main/java/com/flext/nimage/NImageInfo.java
package com.flext.nimage;

import java.util.HashMap;
import java.util.Map;

/**
 * @author : Joe Chan
 * @date : 2024/12/5 16:30
 */
public class NImageInfo {

    public String uri;

    public int imageWidth;

    /// 图片高度，单位px
    public int imageHeight;

    public Map<String, Object> toMap() {
        Map<String, Object> map = new HashMap<>();
        if (uri != null) {
            map.put("uri", uri);
        }
        map.put("imageWidth", imageWidth);
        map.put("imageHeight", imageHeight);
        return map;
    }
}

File: ios/Classes/NImageInfo.swift
import Foundation

public class NImageInfo {
    public var uri: String?
    public var imageWidth: Int?
    public var imageHeight: Int?

    public init() {
    }

    public func toMap() -> [String: Any] {
        var map: Dictionary<String, Any> = [:]
        if let uri = uri {
            map["uri"] = uri
        }
        map["imageWidth"] = imageWidth
        map["imageHeight"] = imageHeight
        return map
    }
}
